src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -217,6 +220,29 @@ var TestRenderer = ReactFiberReconciler({
     setTimeout(fn, 0, {timeRemaining: Infinity});
   },
 
+  mockComponent(component: Fiber, rootContainer: Container) {
+    invariant(
+      component._unmockedType === null,
+      'Trying to mock an already mocked component',
+    );
+    const mockedFn = rootContainer.createComponentMock({
+      type: component.type,
+      props: component.pendingProps,
+    });
+    invariant(
+      typeof mockedFn === 'function',
+      'createComponentMock() must return a function. Found %s instead.',
+      typeof mockedFn,
+    );
+    if (mockedFn !== component.type) {
+      component._unmockedType = component.type;
+      component.type = mockedFn;
+      // force the fiber to be indeterminate so that users can mock a class component
+      // into a functional component and vice versa
+      component.tag = IndeterminateComponent;
```

Comment:
This indicates to me that we're not overriding it soon enough. Maybe we need to do the override earlier?

This also doesn't allow you to mock it to a `string` to render a host node for example.